File: App/app/src/main/java/com/example/econo_misons/database/Envelope.java
package com.example.econo_misons.database;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

@Entity(tableName = "Envelope", foreignKeys = {@ForeignKey(entity = Category.class,
        parentColumns = "ID",
        childColumns = "CAT_ID",
        onDelete = ForeignKey.CASCADE),
        @ForeignKey(entity = PrevisionalBudget.class,
                parentColumns = "YEAR_MONTH",
                childColumns = "PREV_DATE",
                onDelete = ForeignKey.CASCADE)
})
public class Envelope {
    @PrimaryKey
    @NonNull
    @ColumnInfo(name = "PREV_DATE")
    public String dateEnv;

    @ColumnInfo(name = "CAT_ID")
    public int categoryID;

    @ColumnInfo(name = "ENV_SUM")
    public float sumEnv;
}

File: BokuDarjan/app/src/main/java/com/example/bokudarjan/envelope/Envelope.kt
package com.example.bokudarjan.envelope
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "envelope_table")
class Envelope{
    @PrimaryKey(autoGenerate = true)
    var id : Int = 0
    var month : Int = 0;
    var categoryName : String = ""
    var name : String = ""
    var amount : Float = 0F
    var date : String = ""

    constructor()

    constructor(
        categoryName: String,
        name : String,
        amount: Float,
        date: String,
        month: Int
    ) {
        this.categoryName = categoryName
        this.name = name
        this.amount = amount
        this.date = date
        this.month = month
    }

}